test/js/study.test.js

```diff
@@ -0,0 +1,88 @@
+    const { styledText, matchedIndices } = highlightText(text, matchedTerms)
+
+    // Check terms were matched in the correct place
+    expect(matchedIndices).toEqual(matchIndexes)
+    // Check text highlighted properly
+    expect(styledText).toEqual(highlightedText)
+  })
+})
+
+describe('shortenDescription', () => {
+  // 845 characters 133 words
+  const text = 'This study presents an example analysis of an eye (retina) dataset from the Human Cell Atlas (HCA)\
+ Data Coordination Platform (DCP) Project entitled "A single-cell transcriptome atlas of the adult human retina".\
+ It is part of the HCA March 2020 Release (INSERT Link to the DCP page) and showcases HCA single-cell data that were\
+processed with standardized DCP pipelines, further analyzed by Cumulus (LINK), and annotated using published annotations. \
+In this study, you can explore the biological and technical attributes of the analyzed HCA DCP data. Additionally, you can \
+view all HCA Release study pages and search genes across all projects by visiting the Single Cell Portal Release Page. Please \
+note that Release data is not corrected for batch-effects, but is stratified by organ and (in some cases) developmental stage as described below. '
+
+  // For default state where there are no keyword search inquiries
+  it('shortens description for study descriptions > 170 characters', () => {
+    const expectedText = text.slice(0, descriptionWordLimit)
+    const keywordTerms = []
+    const actualText = mount(shortenDescription(text, keywordTerms)).find('.studyDescription').text()
+    expect(expectedText).toEqual(actualText)
+  })
+
+
+  // Matches are within 750 character boundary
+  it('show matches when matches are within 750 character boundary', () => {
+    const expectedText = 'This study presents an example analysis of an eye (retina) dataset from the Human Cell Atlas (HCA) Data Coordination Platform (DCP) Project entitled "A single-cell transcriptome\
+ atlas of the adult human retina". It is part of the HCA March 2020 Release (INSERT Link to the DCP page)\
+ and showcases HCA single-cell data that wereprocessed with standardized DCP pipelines, further analyzed by\
+ Cumulus (LINK), and annotated using published annotations. In this study, you can explore the biological and\
+ technical attributes of the analyzed HCA DCP data. Additionally, you can view all HCA Release study pages and\
+ search genes across all projects by visiting the Single C'
+    const keywordTerms = ['study']
+    const wrapper = mount(shortenDescription(text, keywordTerms))
+    // Find span tag with openingText
+    const openingTextSpan= wrapper.find('span').findWhere(n => n.hasClass('openingText'))
+    // Span tag for opening text should not exist
+    expect(openingTextSpan).toHaveLength(0)
+
+    // Find span with matched text
+    const matchedDescription= wrapper.find('span').findWhere(n => n.hasClass('studyDescription'))
+    expect(matchedDescription).toHaveLength(1)
```

Comment:
I think this test should be checking the number of spans with the class 'highlight' somewhere, since that's the class (formerly id) that drives the highlighting UI.